spec/java/security/SecureRandom.main.lsl

```diff
@@ -0,0 +1,885 @@
+    {
+        var size: int = streamSize as int;
+        if (size > MAX_RANDOM_STREAM_SIZE)
+            size = MAX_RANDOM_STREAM_SIZE;
+
+        val mass: array<int> = _generateRandomIntegerArray(size);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new IntStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = size,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.ints (@target self: SecureRandom, streamSize: long, randomNumberOrigin: int, randomNumberBound: int): IntStream
+    {
+        var size: int = streamSize as int;
+        if (size > MAX_RANDOM_STREAM_SIZE)
+            size = MAX_RANDOM_STREAM_SIZE;
+
+        val mass: array<int> = _generateRandomIntegerArrayWithBounds(size, randomNumberOrigin, randomNumberBound);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new IntStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = size,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.longs (@target self: SecureRandom): LongStream
+    {
+        val mass: array<long> = _generateRandomLongArray(MAX_RANDOM_STREAM_SIZE);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new LongStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = MAX_RANDOM_STREAM_SIZE,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.longs (@target self: SecureRandom, streamSize: long): LongStream
+    {
+        var size: int = streamSize as int;
+        if (size > MAX_RANDOM_STREAM_SIZE)
+            size = MAX_RANDOM_STREAM_SIZE;
+
+        val mass: array<long> = _generateRandomLongArray(size);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new LongStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = size,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.longs (@target self: SecureRandom, randomNumberOrigin: long, randomNumberBound: long): LongStream
+    {
+        val mass: array<long> = _generateRandomLongArrayWithBounds(MAX_RANDOM_STREAM_SIZE, randomNumberOrigin, randomNumberBound);
+        var emptyCloseHandlersList: list<Runnable> = action LIST_NEW();
+
+        result = new LongStreamAutomaton(state = Initialized,
+            storage = mass,
+            length = MAX_RANDOM_STREAM_SIZE,
+            closeHandlers = emptyCloseHandlersList,
+        );
+    }
+
+
+    // within java.util.Random
+    fun *.longs (@target self: SecureRandom, streamSize: long, randomNumberOrigin: long, randomNumberBound: long): LongStream
```

Comment:
What if `randomNumberBound` is **less** than `randomNumberOrigin`?

Reply:
Fixed;